ecs-cli/modules/cli/compose/entity/task/task.go

```diff
@@ -376,12 +379,37 @@ func (t *Task) buildRunTaskInput(taskDefinition string, count int, overrides map
 	}
 
 	if !t.Context().CLIContext.Bool(flags.DisableECSManagedTagsFlag) {
-		runTaskInput.EnableECSManagedTags = aws.Bool(true)
+		enabled, err := isTaskLongARNEnabled(t.Context().ECSClient)
+		if err != nil {
+			return nil, err
+		}
+		if enabled {
+			log.Info("Auto-enabling ECS Managed Tags")
+			runTaskInput.EnableECSManagedTags = aws.Bool(true)
+		}
 	}
 
 	return runTaskInput, nil
 }
 
+func isTaskLongARNEnabled(client ecsclient.ECSClient) (bool, error) {
+	output, err := client.ListAccountSettings(&ecs.ListAccountSettingsInput{
+		EffectiveSettings: aws.Bool(true),
+		Name:              aws.String(ecs.SettingNameTaskLongArnFormat),
+	})
+	if err != nil {
+		return false, err
+	}
+
+	// This should never evaluate to true, unless there is a problem with API
+	// This if block ensures that the CLI does not panic in that case
+	if len(output.Settings) < 1 {
```

Comment:
I may be confused, but doesn't this already exist in a previous PR?

Reply:
The previous PR contained code to check if container instance ARNs were enabled, this one is for task long ARNs. 

HHH and I discussed refactoring this into one method yesterday, but ultimately decided it was okay to keep them separate.